pkg/cache/placeholder_manager.go

```diff
@@ -61,6 +61,8 @@ func NewPlaceholderManager(clients *client.Clients) *PlaceholderManager {
 }
 
 func getPlaceholderManager() *PlaceholderManager {
+	placeholderMgr.Lock()
+	defer placeholderMgr.Unlock()
```

Comment:
add nil check in case `placeholderMgr` is not initiated while calling `getPlaceholderManager()`?
and RLock should be sufficient